packages/stream_chat_flutter/lib/src/message_input/attachment_preview/attachment_preview_screen.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:flutter/material.dart';
+import 'package:stream_chat_flutter/custom_theme/unikon_theme.dart';
+import 'package:stream_chat_flutter/src/message_input/translucent_scafold.dart';
+import 'package:stream_chat_flutter/stream_chat_flutter.dart';
+
+class AttachmentPreviewScreen extends StatefulWidget {
+  const AttachmentPreviewScreen({
+    super.key,
+    required this.effectiveController,
+    required this.attachmentController,
+  });
+  final StreamMessageInputController effectiveController;
+  final StreamAttachmentPickerController attachmentController;
+
+  /// Callback called when the remove button is pressed.
+
+  @override
+  State<AttachmentPreviewScreen> createState() =>
+      _AttachmentPreviewScreenState();
+}
+
+class _AttachmentPreviewScreenState extends State<AttachmentPreviewScreen> {
+  Future<void> _onAttachmentRemovePressed(Attachment attachment) async {
+    final file = attachment.file;
+    final uploadState = attachment.uploadState;
+
+    if (file != null && !uploadState.isSuccess && !isWeb) {
+      await StreamAttachmentHandler.instance.deleteAttachmentFile(
+        attachmentFile: file,
+      );
+    }
+
+    await widget.attachmentController.removeAttachment(attachment);
+    if (widget.attachmentController.value.isEmpty) {
+      Navigator.of(context).pop();
+    }
+
+    if (mounted) setState(() {});
```

Comment:
If possible can't we use ValueNotifier

Reply:
Using the same coding pattern as the package.